### DISCUSSION TITLE
Does heated bed and hotend really affect the accuracy on z?

### DISCUSSION THREAD

**Discussion by Icemasta** [bassamanator/Sovol-SV06-firmware#86] [Q&A]
I've had issues where, even if my z_offset is set up nicely, (and I've done axis twist compensation with heated bed and hotend), my extruder is still too high. Klipper documentation seems to indicate that it's fine as long as it is all done hot or cold. I've been told by some people that it's because I am doing all my tuning while it's hot, instead of cold, that is causing my issues. I figured I'd ask here what I should do.

**Comment by bassamanator** [comment]
I do my z_offset hot. It really shouldn't matter, however, I like to do these things while the environment looks like the environment during print time, that is, the environment is hot!

Try this: a cube to the build plate in your slicer, reduce the height to whatever your first layer height is, and expand x and y to 100mm let's say. Start printing. As you print, babystep your nozzle up/down to perfection. Be sure to hit Save and then to `save_config`. Re-print the part, and you shouldn't have to adjust z_offset again.

What is your z_offset value? Just curious.

**Comment by Icemasta** [comment]
Current it is at z_offset = 1.490, I'll try your method. I'll also enable kamp once I got 5 minutes to do it.

**Reply by bassamanator** [reply]
Something else I would suggest is to use tried, tested, and true slicer settings while you work on this issue. You don't have to become a SuperSlicer user, however, while you work on this problem, consider using the SS config bundle found on this repo.

**Reply by Icemasta** [reply]
I've been using SuperSlicer and PrussaSlicer, I transfered the prove from SuperSlicer pretty much 1:1

**Comment by Manfredo1952** [comment]
in order to establish a good z offset, I am usually slicing and printing this file (fast first layer zigzag), which can be used easily to lower or raise the offset on the fly. I am  a tool to touch the line whilst printing, when to loose, just lower offset 0.01 or 0.02 and so on. Quite easy - https://www.printables.com/model/282044-fast-first-layer-zigzag

**Reply by bassamanator** [reply]
Yeah this part should work nicely as well.

**Comment by Icemasta** [comment]
So used live printing to adjust the z_offset until it looks nice, it got significantly better. I haven't adjusted rotation distance on extruder, so I've been over extruding a tiny bit, I am doing that tomorrow.

That being said, I also did something when calibrating my Z and I just want to know if I did the right thing or the wrong thing.

I did G34 at first but it gave me a very slopped bed mesh, so I figured that the X gantry wasn't level with the bed, so to tune that, I put the Z position just about where the probe stops triggering, disabled all motors, then I moved the hotend to the left, lowered it manually until the probe's light turned off, then I raised it just a tiny bit for the light to turn back on, and then I moved the hotend to the right, did the same thing, and repeated that process about 10 times. This gave me a very level bed mesh.

**But**, that's when Axis twist showed up, so I fixed that using axis twist compensation... which gave me a bed mesh like before, with a slope (because it's what it does). Now, I then printed perfectly, perfectly flat, adjusted the z_offset, but I am curious if my manual adjustment of the X gantry to have it level with the bed caused the axis twist?

Thank you for your time.

**Comment by bassamanator** [comment]
I guess you used this [method](https://www.reddit.com/r/Sovol/comments/113bagl/sv06_xaxis_alignment_using_bed_leveling_sensor/)? I've never used this method, however, I see no reason why it wouldn't work extremely well. You must do it very carefully though.

> But, that's when Axis twist showed up

I don't think you can quickly determine whether you have x-axis twist or not, there are too many factors at play, and only when you go through everything and eleminate all other variables, can you come to the conclusion that you have x-axis twist.

I would perhaps level the gantry against the bed again using the method you used, but watch the video I linked to above. Then re-adjust z_offset, create a new mesh. Then print [this](https://www.reddit.com/r/Sovol/comments/113bagl/sv06_xaxis_alignment_using_bed_leveling_sensor/) to see how the print looks. Remember, if the mesh looks crooked but your prints are coming out well because that's what the mesh is supposed to be there for, then you're fine.

**Reply by Icemasta** [reply]
Alright, thanks a lot. Gonna have to wait sadly, another problem just ran into me. 
![image](https://github.com/bassamanator/Sovol-SV06-firmware/assets/60716357/5153e3a8-de3b-4e57-9312-40dcf338a0ef)

Heater stopped working in the middle of a print. It now does the following behaviour: Will heat up nicely until it reaches ~95% of target temperature, the moment the PWM drops from 100%, it no longer responds to PWM and just stops heating. Eventually it cools down and starts responding again.
